### PR TITLE
Migrate from `moment` to `date-fns`

### LINKED ISSUE #10: Migrate from moment to date-fns
Remove [moment.js](https://momentjs.com/) dependency and use [date-fns](https://date-fns.org/) instead. Moment is great but date-fns is 3.9 times [smaller](https://bundlephobia.com/result?p=date-fns@2.11.1) and [more performant](https://medium.com/@k2u4yt/momentjs-vs-date-fns-6bddc7bfa21e). Also requires to change the library of [material-ui-pickers](https://material-ui-pickers.dev/getting-started/installation#peer-library).

### PR DESCRIPTION
**Pull request by bengry**
## Description
Moment.js has recently [entered maintenance mode](https://momentjs.com/docs/#/-project-status/), and replacing it in this project has been a long-ish standing issue (#10). This PR aims to replace `moment` with `date-fns` completely, with everything that's under this project's direct control.

**Does this PR introduces a breaking change?**
No

### Related Issues
Fixes #10.

## Checklist

- [x] Lint and unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary documentation (if appropriate).

### CODE REVIEW

**Comment by bengry** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by bengry** [comment]
recheckcla

File: package.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
     "ci": "yarn test && yarn test:e2e"
   },
   "dependencies": {
-    "@date-io/moment": "^1.3.13",
+    "@date-io/date-fns": "^2.10.6",
```
Comment (janczizikow) [inline package.json:25]: According to the material-ui pickers [docs](https://material-ui-pickers.dev/getting-started/installation#peer-library), `@date-io/date-fns` should be **v1**

**Review by janczizikow — CHANGES_REQUESTED** [review]
Hey @bengry thanks a lot for the contribution! 

I've noticed that the app crashes when opening a date-picker, it might be something related to using a different version @date-io adapters (see the comment 👆). You can also try test the app while developing using this url: `https://localhost:3000/export/pPEbU07DVOpWNBRMMuA0fyMWgwzppHfEZ`. Or run you can run e2e tests: `yarn test:e2e` (dashboard test suite)

**Comment by janczizikow** [comment]
Also don't worry about the CLA & failing checks something is wrong with the github workflow files :)

**Comment by bengry** [comment]
> Hey @bengry thanks a lot for the contribution!
> 
> I've noticed that the app crashes when opening a date-picker, it might be something related to using a different version @date-io adapters (see the comment 👆). You can also try testing the app while developing using this url: `https://localhost:3000/export/pPEbU07DVOpWNBRMMuA0fyMWgwzppHfEZ`. Or run you can run e2e tests: `yarn test:e2e` (dashboard test suite)

I'll take a look this weekend and fix it, thanks for adding the label in the meantime though!

**Comment by bengry** [comment]
> I've noticed that the app crashes when opening a date-picker, it might be something related to using a different version @date-io adapters (see the comment 👆). You can also try testing the app while developing using this url: `https://localhost:3000/export/pPEbU07DVOpWNBRMMuA0fyMWgwzppHfEZ`. Or run you can run e2e tests: `yarn test:e2e` (dashboard test suite)

Should be fixed in the latest revision I just pushed. App no longer crashed, and the date pickers work as expected as far as I can tell.

I also ran the E2E tests, all passed except for "should login an existing nutri user" (`nutri` spec file). That's because of bad credentials though, which make sense - since I don't have `CYPRESS_username` or `CYPRESS_password` set up locally, and it seems like the request is going to a real server to rest things out (and not a fixture).

**Review by janczizikow — APPROVED** [review]
LGTM 🎉

**Pull request merged**